A5 if-then conditionals (#2051)
deephaven/deephaven-core: demo/web/src/main/notebooks/A. Real-time table ops/A5 If-thens and conditions.md

# If-then and conditions

In our previous notebook, we showed how to group and aggregate time series data. In this notebook, we show how to use if-then statements and conditionals in Deephaven.

Let's start again by simulating some real-time data.

```python
from deephaven.DateTimeUtils import currentTime, expressionToNanos, minus

time_interval = expressionToNanos("1D")
offset = expressionToNanos("1000D")
now = currentTime()

daily_data = create_random_table(time_interval, start_time=minus(now, offset))
```

## Ternaries

Deephaven supports [Java ternaries](https://deephaven.io/core/docs/reference/query-language/control-flow/ternary-if/). We can use these to perform if-else statements on our tables to assign values to a new column based on the if-else evaluation.

```python
daily_data = daily_data.update("IsEven = Number%2 == 0 ? true : false")
```

This query creates a new column `IsEven` in the `daily_data` table.

### Custom methods for ternary conditions

> You may want to use a custom method for your condition for your ternary. To do this, you simply use the method instead of the hardcoded conditional. There is a catch; you do need to cast the method to `(Boolean)` for Deephaven to recognize it properly.

Reviewer: ```suggestion
You may want to use a custom method for your condition for your ternary. To do this, you simply use the method instead of the hardcoded conditional. There is a catch - you need to cast the method to `(Boolean)` for Deephaven to recognize it properly.
```

> Let's use our `is_weekday` method that we used when filtering and decorating our data in a ternary statement to create a new column that tells us if the day of the week is a weekday or not.

Reviewer: ```suggestion
Let's use the `is_weekday` method in a ternary statement to create a new column that tells us if a given day falls during the week or weekend.
```
